src/lib/ivy/cache-compiler-host.ts

```diff
@@ -0,0 +1,166 @@
+import * as ts from 'typescript';
+import * as ng from '@angular/compiler-cli';
+import * as path from 'path';
+import { ensureUnixPath } from '../utils/path';
+
+import { EntryPointNode, fileUrl } from '../ng-package/nodes';
+import { BuildGraph } from '../graph/build-graph';
+import { FileCache } from '../file-system/file-cache';
+import { Node } from '../graph/node';
+import { createHash } from 'crypto';
+import { StylesheetProcessor } from './styles/stylesheet-processor';
+
+export function cacheCompilerHost(
+  graph: BuildGraph,
+  entryPoint: EntryPointNode,
+  compilerOptions: ng.CompilerOptions,
+  moduleResolutionCache: ts.ModuleResolutionCache,
+  stylesheetProcessor?: StylesheetProcessor,
+  sourcesFileCache: FileCache = entryPoint.cache.sourcesFileCache,
+): ng.CompilerHost {
+  const compilerHost = ts.createIncrementalCompilerHost(compilerOptions);
+
+  const getNode = (fileName: string) => {
+    const nodeUri = fileUrl(ensureUnixPath(fileName));
+    let node = graph.get(nodeUri);
+
+    if (!node) {
+      node = new Node(nodeUri);
+      graph.put(node);
+    }
+
+    return node;
+  };
+
+  const addDependee = (fileName: string) => {
+    const node = getNode(fileName);
+    entryPoint.dependsOn(node);
+  };
+
+  return {
+    ...compilerHost,
+
+    // ts specific
+    fileExists: (fileName: string) => {
+      const cache = sourcesFileCache.getOrCreate(fileName);
+      if (cache.exists === undefined) {
+        cache.exists = compilerHost.fileExists.call(this, fileName);
+      }
+      return cache.exists;
+    },
+
+    getSourceFile: (fileName: string, languageVersion: ts.ScriptTarget) => {
+      addDependee(fileName);
+      const cache = sourcesFileCache.getOrCreate(fileName);
+      if (!cache.sourceFile) {
+        cache.sourceFile = compilerHost.getSourceFile.call(this, fileName, languageVersion);
+      }
+      return cache.sourceFile;
+    },
+
+    writeFile: (
+      fileName: string,
+      data: string,
+      writeByteOrderMark: boolean,
+      onError?: (message: string) => void,
+      sourceFiles?: ReadonlyArray<ts.SourceFile>,
+    ) => {
+      if (fileName.endsWith('.d.ts')) {
+        sourceFiles.forEach(source => {
+          const cache = sourcesFileCache.getOrCreate(source.fileName);
+          if (!cache.declarationFileName) {
+            cache.declarationFileName = ensureUnixPath(fileName);
+          }
+        });
+      }
+
+      compilerHost.writeFile.call(this, fileName, data, writeByteOrderMark, onError, sourceFiles);
+    },
+
+    readFile: (fileName: string) => {
+      addDependee(fileName);
+
+      const cache = sourcesFileCache.getOrCreate(fileName);
+      if (cache.content === undefined) {
+        cache.content = compilerHost.readFile.call(this, fileName);
+      }
+      return cache.content;
+    },
+
+    // ng specific
+    moduleNameToFileName: (moduleName: string, containingFile: string) => {
```

Comment:
I don't think this should be implemented for ng-packagr:

https://github.com/angular/angular/blob/de93a7a4bb4fb79098c2d716f3380510ead2f62c/packages/compiler-cli/src/ngtsc/core/api/src/interfaces.ts#L11-L26

Reply:
This one was marked as resolved but the method is still here. Am I wrong in my suggestion that this should not be implemented?